app/controllers/shared_lists_controller.rb

```diff
@@ -0,0 +1,35 @@
+class SharedListsController < ApplicationController
+  before_action :require_user_logged_in!
+
+  def create
+    list_id = shared_params[:list_id]
+    email = shared_params[:email]
+    user_list_shared_with = User.find_by(email: email)
+    @shared_list = SharedList.new(
+      list_id: list_id,
+      user_id: user_list_shared_with&.id
+    )
+    if @shared_list.save
+      respond_to do |format|
+        format.html { redirect_to root_path,
+          notice: "#{List.find_by(id: list_id).name} shared with #{email}"
+        }
+        format.json { render json: {
+          message: "#{List.find_by(id: list_id).name} shared with #{email}"
+          }
+        }
```

Comment:
This is a little hard to read because the calls are spread across lines.  Additionally, convention would state you use do/end instead of curly brackets here because it's spread across lines.  I'm gonna suggest a change here, but this pattern is also lower down.
```suggestion
        format.html do 
          redirect_to root_path, notice: "#{List.find_by(id: list_id).name} shared with #{email}"
        end
        format.json do
          render json: { message: "#{List.find_by(id: list_id).name} shared with #{email}" }
        end
```